docs/dg/dev/guidelines/project-development-guidelines.md

```diff
@@ -23,42 +23,44 @@ There are different strategies developers can use to develop projects. Before st
 
 ## Use custom names
 Use custom names for everything that is added on the project level, like the following:
 - Transfers
 - Transfer properties
-- DB tables
+- DB tables (using project namespace as a prefix is recommended)
 - DB table fields,
 - Modules
 - Classes
-- Interfaces
-- Traits
-- Properties
-- Contants
 
-For example, customize the names by adding the project name.
+For example, customize the names by adding the project name. This will help to avoid conflicts with the Spryker core and other projects.
 
 ## Avoid using, extending, and overriding Private API
-Instead of using, extending, and overriding [Private API](/docs/dg/dev/architecture/module-api/declaration-of-module-apis-public-and-private.html), send a request about the missing endpoints to your Spryker account manager. Spryker offers extension points that allow you to extend via the Public API and helps you to customize the application. We recommend to use the extension points instead of overriding Private API.
+Instead of using, extending, and overriding [Private API](/docs/dg/dev/architecture/module-api/declaration-of-module-apis-public-and-private.html) (everything that is not a Public API), send a request about the missing endpoints to your Spryker account manager. Spryker offers extension points that allow you to extend via the Public API and helps you to customize the application. We recommend to use the extension points instead of overriding Private API.
 
-## Avoid using unsupported types
-Avoid using unsupported types as constructor arguments of a plugin.
+Extending Private API is still possible, but core development teams are not considering changing those classes as any BC break as long as a Public API is not affected. So you can get a critical error with your next minor update.
+As a recommendation to avoid such issues, we recommend to not extend classes, but implement new one based on the same interface. In this case you still can get a minor update where this Private API was changed, but it much easier to fix it. This approach is not valid if you want to extend e.g. Facade, because you will be forced to re-implement all the original methods.
 
-The [supported types](/docs/dg/dev/guidelines/keeping-a-project-upgradable/upgradability-guidelines/single-plugin-argument.html#problem-description) are: null, bool, integer, float, string and object.
-
-## Keep modules up to date
-During development and after going live, we recommend checking for security updates of external and Spryker dependencies on a regular basis.
-
-Additionally, we recommend keeping modules up to date in general.
+If there is an error, or you want to submit a fix or a new functionality by yourself, you can create a pull request to the required repository. It will be reviewed by the Spryker team and, if it is accepted, it will be merged into the core via our release process. Be advised that such PRs will be checked as a core contribution and should be prepared accordingly.
```

Comment:
```suggestion
If you believe that you found a bug in Spryker module, or you want to submit a fix, new functionality by yourself, you can create a pull request to the corresponding module repository. It will be reviewed by the Spryker team and, if it is accepted, it will be merged into the core via our release process. Be advised that such PRs will be checked as a Core contribution and should be prepared accordingly.
```

Reply:
Can we please post a link to the Core contribution guidelines here?